### ISSUE TITLE
Getting Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.20.1:test (default-test) on project TestSetup: There are test failures.

### ISSUE THREAD

**Issue by Navsony** [cbeust/testng#1693]
### TestNG Version
> 6.14.2

I am using the following components.

TestNG -6.14.2
jdk -9.0.1
maven-compiler-plugin - 3.7.0
maven-surefire-plugin - 2.20.1 

### Actual behavior

Maven build getting failed due to Assertion failure in the one of the tests of the test suite. with following error.

Getting Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project TestSetup: There are test failures.

```
INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 04:08 min
[INFO] Finished at: 2018-02-07T15:51:00+05:30
[INFO] Final Memory: 13M/44M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project TestSetup: There are test failures.
[ERROR]
[ERROR] Please refer to D:\TestWorkspace\TestSetup\target\surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

### Is the issue reproductible on runner?
- [YES ] Maven
- [YES ] Eclipse

### Test case sample
> Any test case with Assertion failure.

Please help on this.

**Comment by juherr** [comment]
It looks like a surefire issue. Maybe you should first open an issue there: https://issues.apache.org/jira/projects/SUREFIRE. (fyi @tibor17)

BTW, could you share a minimal project that shows the issue?

**Comment by Tibor17** [comment]
Can you check it our with `2.21.0-SNAPSHOT`. Do not download it from Nexus. Clone the branch `SUREFIRE-1463`, build it `mvn install -DskipTests` and run your project `offline` with `mvn -o test`.
I am pretty sure that this would work.

**Comment by sonee123** [comment]
Hii Tibor,

mvn install -DskipTests it's  really work fine.

Thank u so much.

**Comment by krmahadevan** [comment]
Closing this off because this issue is not related to TestNG.

**Comment by Tibor17** [comment]
@jayanth-ganganahalli 
Hard for me. I do not see your project. Did you use the latest Surefire 2.22.0?

**Comment by ashishu74** [comment]
I am also having same error mentioned above. Can any one suggest what could be solution for this.

**Comment by Tibor17** [comment]
@juherr 
@Navsony 
@sonee123 
@krmahadevan 
I do not understand what I should do and why because here is assertion error which is normal because developer/you broke your code and therefore build error. See the first comment - issue. It's practically empty.

**Comment by krmahadevan** [comment]
@Tibor17 - I am part of the TestNG dev group.. so I don't have an issue here.

@ashishu74 - If you are having a similar error, please provide the following

1. A simple standalone project that can be used to reproduce the issue (Its better if you could help provide a complete maven project) Please have the project not use any 3rd party libraries but just use core java alone.

**Comment by ahmad92hassan** [comment]
Thanks a lot sonee123
its worked

**Comment by manideepallala** [comment]
After installing below plugin, isuue was resolved

			<plugin>
    			<groupId>org.apache.maven.plugins</groupId>
    			<artifactId>maven-surefire-plugin</artifactId>
   			    <version>2.19.1</version>
   				<configuration>
        		<testFailureIgnore>true</testFailureIgnore>
    			</configuration>
			</plugin>

**Comment by kandy87** [comment]
I had the same issue and adding "-DskipTests" to script fixed the issue. But I'm not sure whether it is the correct fix. Using 2.22.1 surefire fixed the issue (as @krmahadevan mentioned) with test cases execution.

**Comment by LeonidAlekseenko** [comment]
thank you it works!

**Comment by BoHuang2018** [comment]
> I had the same issue and adding "-DskipTests" to script fixed the issue. But I'm not sure whether it is the correct fix. Using 2.22.1 surefire fixed the issue (as @krmahadevan mentioned) with test cases execution.
> 
> Thnx guys,

To my error:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.21.0:test (default-test) on project my-working-tests: There are test failures.
```

The error was solved by adding the `<version>2.22.1</version>` to the `pom.xml` fie.

Old code (raised the error):
```
<plugin>
          <groupId>org.apache.maven.plugins</groupId>
          <artifactId>maven-surefire-plugin</artifactId>
          <configuration>
           .......
          </configuration>
</plugin>
```

Updated code (solved the error):
```
<plugin>
          <groupId>org.apache.maven.plugins</groupId>
          <artifactId>maven-surefire-plugin</artifactId>
          <version>2.22.1</version>
          <configuration>
           .......
          </configuration>
</plugin>
```

**Comment by yPratikhi** [comment]
![Uploading Screenshot (450).png…]()
> Can you check it our with `2.21.0-SNAPSHOT`. Do not download it from Nexus. Clone the branch `SUREFIRE-1463`, build it `mvn install -DskipTests` and run your project `offline` with `mvn -o test`. I am pretty sure that this would work.

![Screenshot (451)](https://github.com/testng-team/testng/assets/73225825/783f45ff-488f-43c1-a9b6-46860216ad90)
